autosubmit/job/job.py

```diff
@@ -222,74 +213,73 @@ def __init__(self, name=None, job_id=None, status=None, priority=None, loaded_da
         self._long_name = None
         self.date_format = ''
         self.type = Language.BASH
-        self.undefined_variables = None
+        self.undefined_variables: list[str] = []
         self.log_retries = 5
         self.id = job_id
-        self.file = None
-        self.additional_files = []
+        self.file: str = ''
+        self.additional_files: list[str] = []
         self.executable = None
         self._local_logs = ('', '')
         self._remote_logs = ('', '')
         self.script_name = self.name + ".cmd"
-        self.stat_file = f"{self.script_name[:-4]}_STAT_"
+        self.stat_file: Optional[str] = f"{self.script_name[:-4]}_STAT_"
         self._status = None
         self.status = status
         self.prev_status = status
         self.new_status = status
         self.priority = priority
-        self._parents = set()
-        self._children = set()
+        self._parents: set = set()
+        self._children: set = set()
         self._fail_count = 0
         """Number of failed attempts to run this job. (FAIL_COUNT)"""
-        self.expid: str = name.split('_')[0]
+        self.expid: Optional[str] = name.split('_')[0]
         self._tmp_path = os.path.join(
             BasicConfig.LOCAL_ROOT_DIR, self.expid, BasicConfig.LOCAL_TMP_DIR)
         self._log_path = Path(f"{self._tmp_path}/LOG_{self.expid}")
-        self._platform = None
         self.check = 'true'
         self.check_warnings = False
         self.packed = False
-        self.hold = False  # type: bool
+        self.hold = False
         self.distance_weight = 0
         self.level = 0
         self._export = "none"
-        self._dependencies = []
+        self._dependencies: list = []
         self.running = None
-        self.start_time = None
-        self.ext_header_path = None
-        self.ext_tailer_path = None
-        self.edge_info = dict()
+        self.start_time: Optional[datetime.datetime] = None
+        self.ext_header_path: str = ''
+        self.ext_tailer_path: str = ''
+        self.edge_info: dict = {}
         self.total_jobs = None
         self.max_waiting_jobs = None
-        self.exclusive = ""
+        self.exclusive: Optional[str] = ""
         self._retrials = 0
         # internal
-        self.current_checkpoint_step = 0
-        self.max_checkpoint_step = 0
-        self.reservation = ""
+        self.current_checkpoint_step: int = 0
+        self.max_checkpoint_step: Optional[int] = 0
+        self.reservation: Optional[str] = ""
         self.delete_when_edgeless = False
         # hetjobs
-        self.het = None
+        self.het: dict[Any, Any]
```

Comment:
I think while harmless, here we are setting the variable as `None` but declaring as `dict`? Maybe it should be initialized/